packages/hash/api/src/db/postgres/link/sql/links.util.ts

```diff
@@ -0,0 +1,304 @@
+import { sql, ValueExpressionType } from "slonik";
+
+import {
+  DBLinkVersionRow,
+  linkVersionsColumnNames,
+  insertLinkVersionRow,
+} from "./link_versions.util";
+import { Connection } from "../../types";
+import { DBLink } from "../../../adapter";
+import { mapColumnNamesToSQL } from "../../util";
+import { insertIncomingLinkRow } from "./incoming_links.util";
+
+export const linksColumnNames = [
+  "link_id",
+  "path",
+  "source_account_id",
+  "source_entity_id",
+  "applied_to_source_at",
+  "applied_to_source_by",
+  "removed_from_source_at",
+  "removed_from_source_by",
+  "destination_account_id",
+  "destination_entity_id",
+  "destination_entity_version_id",
+];
+
+export const linksColumnNamesSQL = mapColumnNamesToSQL(linksColumnNames);
+
+export type DBLinkRow = {
+  link_id: string;
+  path: string;
+  index: number | null;
```

Comment:
Perhaps this is blocking

Reply:
The `links` PG table does not, but the `DBLink` type no longer represents the contents of just that table (similar to how the `DBEntity` type doesn't just represent the contents of the `entities` PG table, but combines columns from `entities`, `entity_versions` & `entity_types`).

The reasoning behind this is because I can't think of a scenario where we want to pass a link without it's index in the response of a db adapter method, so might as well be able to treat a `DBLink` as a combination of all its properties we store in the db (even if they're in separate tables)

Reply:
Got it, no worries then!